### PR TITLE
Add future delegation mapping and adapt `DELEGS` STS rule

### PR DESCRIPTION
**Pull request by mgudemann**
Incorporates changes due to #565 #579

### CODE REVIEW

File: shelley/chain-and-ledger/formal-spec/delegation.tex [hunk 1]
```diff
@@ -954,13 +955,16 @@ \subsection{Delegation and Pool Combined Rules}
     \inference[Seq-delg-ind]
     {
       \var{c}\in\DCertDeleg \Rightarrow \fun{dpool}~{c} \in \dom \var{stpools} \\
-        ptr \leteq (\var{slot},~\var{txIx},~\mathsf{len}~\Gamma - 1) \\~\\
+      ptr \leteq (\var{slot},~\var{txIx},~\mathsf{len}~\Gamma - 1) \\~\\
+      % TODO should that be "len \Gamma" ? at this point \Gamma could be the
```
Comment (JaredCorduan) [inline shelley/chain-and-ledger/formal-spec/delegation.tex]: in the spec there is a case distinction separating the empty list, which is handled by `Seq-delg-base`. I'm not sure what is the best thing to do in the STS framework where the case split doesn't happen here.

Comment (mgudemann) [reply shelley/chain-and-ledger/formal-spec/delegation.tex]: yes, but here we construct the `ptr` which is used in the `DELPL` rule which takes a single certificate `c`. So wouldn't that also compute `len Γ -1` in the case of `∅; c` ?

Comment (JaredCorduan) [reply shelley/chain-and-ledger/formal-spec/delegation.tex]: ah yes, the full list is `Gamma;c`, not `Gamma`, you are correct.

Comment (mgudemann) [reply shelley/chain-and-ledger/formal-spec/delegation.tex]: ok, I'll change it then in the formal spec

Comment (JaredCorduan) [reply shelley/chain-and-ledger/formal-spec/delegation.tex]: thank you!

**Pull request merged**